### DISCUSSION TITLE
Cannot install Homebrew on iMac M1, macOS Sonoma 14.2.1

### DISCUSSION THREAD

**Discussion by DeDunga** [Homebrew/discussions#5068] [Getting started]
### Output of `brew config`

```shell
I want to install Git. According to https://git-scm.com/download/mac, 
Homebrew is a prerequisite to install Git. 
My ststem:
1. Desktop Apple M1 
2. macOS Sonoma 14.2.1 
3. Xcode version 15.2
I am using the following bash command to install homebrew (got it from brew.sh)
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
```

### Output of `brew doctor`

```shell
The output I get from the bash command is:
==> Checking for `sudo` access (which may request your password)...
Password:
==> This script will install:
/opt/homebrew/bin/brew
/opt/homebrew/share/doc/homebrew
/opt/homebrew/share/man/man1/brew.1
/opt/homebrew/share/zsh/site-functions/_brew
/opt/homebrew/etc/bash_completion.d/brew
/opt/homebrew

Press RETURN/ENTER to continue or any other key to abort:
==> /usr/bin/sudo /usr/sbin/chown -R sanjeevjoshi:admin /opt/homebrew
==> Downloading and installing Homebrew...
xcrun: error: unable to load libxcrun (dlopen(/Library/Developer/CommandLineTools
/usr/lib/libxcrun.dylib, 0x0005): tried: '/Library/Developer/CommandLineTools/usr/lib/libxcrun.dylib' 
(mach-o file, but is an incompatible architecture (have 'x86_64', need '')), '/System/Volumes/Preboot/Cryptexes/OS/Library/Developer/CommandLineTools
/usr/lib/libxcrun.dylib' (no such file), '/Library/Developer/CommandLineTools/
usr/lib/libxcrun.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need ''))).
Failed during: /usr/bin/git -c init.defaultBranch=master init --quiet
```

### Description of issue

I obviously don't know what is 'brew config' and 'brew doctor'. I want to install Homebrew and then use it to install Git. On the Git website, it is mentioned that "Apple ships binary package of Git with Xcode." Does that have something to do with failed Homebrew installation? Can I install Git without Homebrew? 
Your help will be greatly appreciated.

**Comment by Bo98** [comment]
Looks like `/Library/Developer/CommandLineTools/usr/lib/libxcrun.dylib` is corrupt. Try `xcode-select --install`. If it doesn't do anything, you may have to remove your existing `/Library/Developer/CommandLineTools` folder first.

**Comment by hemanshukumar** [comment]
Hi, I have a (at least superficially) similar problem, where I get the following message while trying to install HomeBrew:

`Can't load '/Library/Perl/5.30/darwin-thread-multi-2level/auto/Digest/SHA/SHA.bundle' for module Digest::SHA: dlopen(/Library/Perl/5.30/darwin-thread-multi-2level/auto/Digest/SHA/SHA.bundle, 0x0001): tried: '/Library/Perl/5.30/darwin-thread-multi-2level/auto/Digest/SHA/SHA.bundle' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/Library/Perl/5.30/darwin-thread-multi-2level/auto/Digest/SHA/SHA.bundle' (no such file), '/Library/Perl/5.30/darwin-thread-multi-2level/auto/Digest/SHA/SHA.bundle' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')) at /System/Library/Perl/5.30/darwin-thread-multi-2level/DynaLoader.pm line 197.
 at /usr/bin/shasum5.30 line 25. `

I raised this issue [here](https://github.com/orgs/Homebrew/discussions/5048) but haven't yet been able to get it resolved. Any suggestions on what to do?

Sorry to barge in here, but just thought it might be related -- I'm also on Sonoma 14.2.1 and the error is similar, even if in a completely different file.

**Comment by DeDunga** [comment]
I was revealed to me that Git comes bundled with Xcode, hence no need of Homebrew. Earlier, I installed Xcode version 15.2 from the Apple developer site. I have no idea what went wrong; some/many files were corrupted. Then I reinstalled Xcode, same version, 15.2, from the App store and it seems to be well-behaved. I can't explain why it is so, but I was happy to get an output from the command git --version -> git version 2.39.3 (Apple Git-145). Thank you Bo98. 'hemanshukumar': I have not used Homebrew in quite a while.
